### PR TITLE
Check Signature on Worker or Reputer Payload Matches Inferer/Forecaster/Reputer inside Bundle

### PR DESCRIPTION
**Pull request by relyt29**
## Purpose of Changes and their Description

This PR prevents an attack where someone uploading a WorkerPayload or ReputerPayload could upload a payload that has a valid signature upon a bundle, but the inferer, forecaster, or reputer listed as responsible for the data of the bundle, was not actually the person who signed the bundle.

E.g. Alice uploads a WorkerPayload that has a signature from alice that is valid. But inside the signed Bundle, the bundle says that Bob said that the inference was 400 - even though Bob didn't sign the actual payload.

## Link(s) to Ticket(s) or Issue(s) resolved by this PR

This was discovered independently by me while working on #548 and does not have a ticket.

## Are these changes tested and documented?

Our existing test suite at least covers the happy path, when actors are well behaved.

This PR adds the following four tests to cover authorization checks:
* `TestMsgInsertWorkerPayloadWorkerNotMatchSignature`
* `TestMsgInsertWorkerPayloadInfererNotMatchSignature`
* `TestMsgInsertWorkerPayloadForecasterNotMatchSignature`
* `TestMsgInsertReputerPayloadReputerNotMatchSignature`

No documentation changes.

This PR has been added to the `Unreleased` section of `CHANGELOG.md`?

### CODE REVIEW

**Comment by relyt29** [comment]
NOTE: I am setting the base branch on this PR to #542 right now because I want to show a minimal diff, but this should be merged into `dev` not 542, so that the commit history shows it separately, given it's an important security fix

File: x/emissions/types/worker_data_bundle.go [hunk 1]
```diff
@@ -3,58 +3,83 @@ package types
...
-	pubkey := secp256k1.PubKey(pk)
-
 	src := make([]byte, 0)
 	src, _ = bundle.InferenceForecastsBundle.XXX_Marshal(src, true)
 	if !pubkey.VerifySignature(src, bundle.InferencesForecastsBundleSignature) {
-		return errors.Wrap(sdkerrors.ErrInvalidRequest, "signature verification failed")
+		return errorsmod.Wrap(sdkerrors.ErrUnauthorized, "signature verification failed")
+	}
+	// Source: https://docs.cosmos.network/v0.46/basics/accounts.html#addresses
+	if sdk.AccAddress(pubkey.Address().Bytes()).String() != bundle.Worker {
```
[the remark this answers is not in the document]
Comment (relyt29) [reply x/emissions/types/worker_data_bundle.go:81]: I check `msg.ReputerValueBundle.ValueBundle.Reputer` in `x/emissions/keeper/msgserver/msg_server_reputer_payload.go` but let me know if theres anything else we should check

[the remark this answers is not in the document]
Comment (relyt29) [reply x/emissions/types/worker_data_bundle.go:81]: yes that's implemented in this PR already

https://github.com/allora-network/allora-chain/blob/ebfd5bba250b6529c771d9a7669217d5d5f1f204/x/emissions/types/reputer_value_bundle.go#L29-L31

**Comment by kpeluso** [comment]
changed the base because when @RedBird96 's PR is merged to dev, this will automatically have its base branch changed to dev. we just should eb sure to not merge until @RedBird96 's is!

**Review by kpeluso — APPROVED** [review]

**Pull request merged**